test/unit/OracleDDLHelper.spec.ts

```diff
@@ -1,23 +1,105 @@
+            getConnection: async () => Promise.resolve(stubConnection),
+            releaseConnection: async () => Promise.resolve()
+        };
+
+        return {
+            pool
+        };
+    };
+
     describe('getDDLQuery', () => {
 
-        it('should return a properly formatted query for all specified tables', async () => {
-            const helper = new OracleDDLHelper();
-            const query = helper.getDDLQuery(['table1']);
+        it('should return a properly formatted query for a single table', async () => {
+            const ctx = createTestContext();
+            const helper = new OracleDDLHelper(ctx.pool);
+            const query = await helper.getDDLQuery(['table1']);
+
+            expect(query).to.equal(`SELECT 0 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL('TABLE', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = 'table1' `);
+        });
+
+        it('should return a properly formatted query for multiple unrelated tables', async () => {
+            const ctx = createTestContext();
+            const helper = new OracleDDLHelper(ctx.pool);
+            const query = await helper.getDDLQuery(['table1', 'table2', 'table3']);
+
+            expect(query).to.contain('SELECT 0 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table1\'');
+            expect(query).to.contain('SELECT 1 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table2\'');
+            expect(query).to.contain('SELECT 2 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table3\'');
+        });
+
+        it('should return a properly formatted query for multiple related tables', async () => {
+            const ctx = createTestContext([['table1', 'table2'], ['table3', 'table1']]);
+            const helper = new OracleDDLHelper(ctx.pool);
+            const query = await helper.getDDLQuery(['table1', 'table2', 'table3']);
+
+            expect(query).to.contain('SELECT 0 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table2\'');
+            expect(query).to.contain('SELECT 1 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table1\'');
+            expect(query).to.contain('SELECT 2 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'TABLE\', TABLE_NAME, OWNER) as ddl FROM ALL_TABLES WHERE TABLE_NAME = \'table3\'');
+        });
+
+        it('should return a properly formatted query for multiple related and unrelated tables', async () => {
+            const ctx = createTestContext([['table1', 'table2']]);
+            const helper = new OracleDDLHelper(ctx.pool);
+            const query = await helper.getDDLQuery(['table3', 'table1', 'table2']);
```

Comment:
In cases like this it really doesn't matter where table3 lands in the priority list as there are no dependencies on it. Placing  it at the end of the 'valid table names' arg here would land it at the end of the prioritized list without any side effects.